src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ClusterDnsCacheSpec.scala

```diff
@@ -1,142 +1,72 @@
-    ClusterDnsCache.HostToIp = Map.empty
-
     // save the clusters to the db
-    cluster1.save() shouldEqual cluster1
-    cluster2.save() shouldEqual cluster2
-    cluster3.save() shouldEqual cluster3
-
-    // maps should be populated
-    eventually {
-      actorRef.underlyingActor.ProjectNameToHost shouldBe Map(
-        (project, name1) -> ClusterReady(Host(s"${cluster1.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org")),
-        (project, name2) -> ClusterNotReady,
-        (project, name3) -> ClusterPaused
-      )
-      ClusterDnsCache.HostToIp shouldBe Map(
-        Host(s"${cluster1.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org") -> cluster1.dataprocInfo.hostIp.get
-      )
-    }
-
-    // calling GetByProjectAndName should return the correct responses
-    (actorRef ? GetByProjectAndName(project, name1)).futureValue shouldBe ClusterReady(Host(s"${cluster1.dataprocInfo.googleId.get.toString}.jupyter.firecloud.org"))
-    (actorRef ? GetByProjectAndName(project, name2)).futureValue shouldBe ClusterNotReady
-    (actorRef ? GetByProjectAndName(project, name3)).futureValue shouldBe ClusterPaused
-    (actorRef ? GetByProjectAndName(project, ClusterName("bogus"))).futureValue shouldBe ClusterNotFound
-    (actorRef ? GetByProjectAndName(project, name0)).futureValue shouldBe ClusterNotFound
-  }
-
-  it should "return a ready cluster after a ProcessReadyCluster message" in isolatedDbTest {
-    val actorRef = TestActorRef[ClusterDnsCache](ClusterDnsCache.props(proxyConfig, DbSingleton.ref))
-
-    // init maps to empty
-    actorRef.underlyingActor.ProjectNameToHost = Map.empty
-    ClusterDnsCache.HostToIp = Map.empty
-
-    // save the clusters to the db
-    cluster1.save() shouldEqual cluster1
-    cluster2.save() shouldEqual cluster2
-    cluster3.save() shouldEqual cluster3
-
-    // we have not yet executed a DNS cache refresh cycle
+    clusterBeingCreated.save() shouldEqual clusterBeingCreated
+    runningCluster.save() shouldEqual runningCluster
+    stoppedCluster.save() shouldEqual stoppedCluster
 
-    (actorRef ? GetByProjectAndName(project, name1)).futureValue shouldBe ClusterNotFound
-    (actorRef ? GetByProjectAndName(project, name2)).futureValue shouldBe ClusterNotFound
-    (actorRef ? GetByProjectAndName(project, name3)).futureValue shouldBe ClusterNotFound
+    // We test the projectClusterToHostStatus cache before the hostToIp map.
+    // This replicates how the proxy accesses these maps as well.
+    // projectClusterToHostStatus read updates the HostToIP map.
+    eventually { clusterDnsCache.cache.get(cacheKeyForClusterBeingCreated).futureValue shouldEqual HostNotReady }
+    eventually { clusterDnsCache.cache.get(cacheKeyForRunningCluster).futureValue shouldEqual HostReady(runningClusterHost) }
+    eventually { clusterDnsCache.cache.get(cacheKeyForStoppedCluster).futureValue shouldEqual HostPaused }
 
-    // add one cluster to cache immediately without waiting for the cycle
```

Comment:
Using `CacheStats` (enabled at the time of cache creation via `recordStats` in `ClusterDnsCache.scala`) to make sure the expected number of cache loadings, misses and evictions happen throughout the test. I was motivated to do this particularly because of the `500` errors we saw recently that looked like a DNS cache. 

From what I read, enabling stats recording shouldn't cause a noticeable performance penalty.